service/controller/cluster.go

```diff
@@ -223,6 +260,15 @@ func newClusterResources(config ClusterConfig) ([]resource.Interface, error) {
 		// clusterConfigMapResource is executed before the app resource so the
 		// app CRs are accepted by the validation webhook.
 		clusterConfigMapResource,
+	}
+
+	if config.Provider == "openstack" {
```

Comment:
For now, we only want this resource on openstack. The secret would be empty on other providers.